src/weights.jl

```diff
@@ -22,7 +22,18 @@ values(wv::AbstractWeights) = wv.values
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 
-Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
+function getindex(A::AbstractWeights{T}, i...) where {T}
+    # @boundscheck checkbounds(A, I...)
```

Comment:
This should be uncommented, and the function should be marked with `@propagate_inbounds`.

Reply:
hmm, tells me
```
ERROR: LoadError: LoadError: UndefVarError: @propagate_inbounds not defined
```
I have julia 0.6.2.

Reply:
additionally, now i remember why i commented this out. wiht the boundscheck line, i see

```julia
julia> StatsBase.Weights([1; 2; 3])
3-element StatsBase.Weights{Int64,Int64,Array{Int64,1}}:
Error showing value of type StatsBase.Weights{Int64,Int64,Array{Int64,1}}:
ERROR: MethodError: no method matching start(::UniformScaling{Int64})
Closest candidates are:
  start(::SimpleVector) at essentials.jl:258
  start(::Base.MethodList) at reflection.jl:560
  start(::ExponentialBackOff) at error.jl:107
  ...
Stacktrace:
 [1] append_any(::Tuple{StatsBase.Weights{Int64,Int64,Array{Int64,1}}}, ::Vararg{Any,N} where N) at ./essentials.jl:169
 [2] getindex(::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::Int64, ::Int64) at /Users/74097/.julia/v0.6/StatsBase/src/weights.jl:31
 [3] isassigned(::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::Int64, ::Int64, ::Vararg{Int64,N} where N) at ./abstractarray.jl:222
 [4] alignment(::IOContext{Base.Terminals.TTYTerminal}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::Base.OneTo{Int64}, ::Base.OneTo{Int64}, ::Int64, ::Int64, ::Int64) at ./show.jl:1353
 [5] print_matrix(::IOContext{Base.Terminals.TTYTerminal}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::String, ::String, ::String, ::String, ::String, ::String, ::Int64, ::Int64) at ./show.jl:1482
 [6] print_matrix(::IOContext{Base.Terminals.TTYTerminal}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::String, ::String, ::String) at ./show.jl:1455
 [7] #showarray#263(::Bool, ::Function, ::IOContext{Base.Terminals.TTYTerminal}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}, ::Bool) at ./show.jl:1704
 [8] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}) at ./REPL.jl:122
 [9] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::StatsBase.Weights{Int64,Int64,Array{Int64,1}}) at ./REPL.jl:125
 [10] display(::StatsBase.Weights{Int64,Int64,Array{Int64,1}}) at ./multimedia.jl:218
 [11] eval(::Module, ::Any) at ./boot.jl:235
 [12] print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:144
 [13] print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:129
 [14] (::Base.REPL.#do_respond#16{Bool,Base.REPL.##26#36{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:646

```

Reply:
That's because `I...` should be `i`.

For `@propagate_inbounds`, it's apparently not exported from `Base`, but anyway it seems `@inline` is appropriate here instead.